planner/core/task.go

```diff
@@ -435,6 +436,10 @@ func (p *PhysicalHashJoin) GetCost(lCnt, rCnt float64) float64 {
 	probeCost /= float64(p.Concurrency)
 	// Cost of additional concurrent goroutines.
 	cpuCost += probeCost + float64(p.Concurrency+1)*sessVars.ConcurrencyFactor
+	// Cost of traveling the hash table when building the hash table from the outer table
+	if p.UseOuterToBuild {
+		cpuCost += innerCnt * sessVars.CPUFactor / float64(p.Concurrency)
```

Comment:
Do we take the probe cost into consideration before this commit?

Reply:
Yes, code lays at Line:429
```
probeCost := numPairs * sessVars.CPUFactor
```

Reply:
Actually we do not need line 440~441?

Reply:
Yes， traversing the hash table to find missing cases and output missing matched results.